src/quo2/components/bottom_sheet/view.cljs

```diff
@@ -0,0 +1,218 @@
+        master-state         (animated/use-value (:undetermined gesture-handler/states))
+        tap-state            (animated/use-value 0)
+        manual-open          (animated/use-value 0)
+        manual-close         (animated/use-value 0)
+        offset               (animated/use-value 0)
+        drag-over            (animated/use-value 1)
+        clock                (animated/use-clock)
+        tap-gesture-handler  (animated/use-gesture {:state tap-state})
+        on-master-event      (animated/use-gesture
+                              {:translationY master-translation-y
+                               :state        master-state
+                               :velocityY    master-velocity-y})
+        on-body-event        on-master-event
+        sheet-height         (min max-height
+                                  (+ styles/border-radius @height))
+
+        open-snap-point  (animated/use-value 0)
+        close-snap-point 0
+        on-close         (fn []
+                           (when @visible
+                             (reset! visible false)
+                             (reset! height 0)
+                             (when on-cancel
+                               (on-cancel))))
+        close-sheet      (fn []
+                           (animated/set-value manual-close 1))
+        on-snap          (fn [pos]
+                           (when (= close-snap-point (aget pos 0))
+                             (on-close)))
+        interrupted      (animated/and* (animated/eq master-state (:began gesture-handler/states))
+                                        (animated/clock-running clock))
+        translate-y      (react/use-memo
+                          (fn []
+                            (animated/with-easing
+                              {:value          (animated/cond* (animated/less-or-eq master-translation-y 0)
+                                                               (animated/divide master-translation-y 2)
+                                                               master-translation-y)
+                               :velocity       master-velocity-y
+                               :offset         offset
+                               :state          master-state
+                               :animation-over drag-over
+                               :snap-points    [open-snap-point close-snap-point]}))
+                          [])
+        opacity          (react/use-memo
+                          (fn []
+                            (animated/cond*
+                             open-snap-point
+                             (animated/interpolate
+                              translate-y
+                              {:inputRange  [(animated/multiply open-snap-point opacity-coeff) 0]
+                               :outputRange [1 0]
+                               :extrapolate (:clamp animated/extrapolate)})))
+                          [])]
```

Comment:
Can you kindly elaborate why we use useMemo if we don't have a dependency array?

Reply:
I copied this code over from the previous bottom sheet component. no changes here.

Reply:
Kindly read this

https://matheus.ro/2017/12/11/clean-code-boy-scout-rule/

Reply:
Do we want to rewrite and improve the whole bottom sheet? @flexsurfer 

My goal was to just duplicate and/or move it over to quo2. What do you guys think? @flexsurfer @ibrkhalil

Reply:
I'll create an issue for it and assign to myself
Thanks for the refactors! 😄